pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1180,6 +1182,162 @@ func TestPreFilterState(t *testing.T) {
+				SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Obj(), nil, nil, nil, []string{"foo"}).
+				Obj(),
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label("zone", "zone1").Label("node", "node-a").Obj(),
+				st.MakeNode().Name("node-b").Label("zone", "zone1").Label("node", "node-b").Obj(),
+				st.MakeNode().Name("node-x").Label("zone", "zone2").Label("node", "node-x").Obj(),
+				st.MakeNode().Name("node-y").Label("zone", "zone2").Label("node", "node-y").Obj(),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("p-a1").Node("node-a").Label("foo", "a").Obj(),
+				st.MakePod().Name("p-a2").Node("node-a").Label("foo", "a").Obj(),
+				st.MakePod().Name("p-b1").Node("node-b").Label("foo", "a").Obj(),
+				st.MakePod().Name("p-y1").Node("node-y").Label("foo", "a").Obj(),
+				st.MakePod().Name("p-y2").Node("node-y").Label("foo", "a").Obj(),
+			},
+			want: &preFilterState{
+				Constraints: []topologySpreadConstraint{
+					{
+						MaxSkew:            1,
+						TopologyKey:        "zone",
+						Selector:           mustConvertLabelSelectorAsSelector(t, st.MakeLabelSelector().Label("foo", "a").Obj()),
+						MinDomains:         1,
+						NodeAffinityPolicy: v1.NodeInclusionPolicyHonor,
+						NodeTaintsPolicy:   v1.NodeInclusionPolicyIgnore,
+					},
+				},
+				TpKeyToCriticalPaths: map[string]*criticalPaths{
+					"zone": {{"zone2", 2}, {"zone1", 3}},
+				},
+				TpPairToMatchNum: map[topologyPair]int{
+					{key: "zone", value: "zone1"}: 3,
+					{key: "zone", value: "zone2"}: 2,
+				},
+			},
+			enableMatchLabelKeys: true,
+		},
+		{
+			name: "matchLabelKeys ANDed with LabelSelector when there are different values for the same key in pod.labels and LabelSelector",
+			pod: st.MakePod().Name("p").Label("foo", "a").
+				SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().In("foo", []string{"b"}).Obj(), nil, nil, nil, []string{"foo"}).
+				Obj(),
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label("zone", "zone1").Label("node", "node-a").Obj(),
+				st.MakeNode().Name("node-b").Label("zone", "zone1").Label("node", "node-b").Obj(),
+				st.MakeNode().Name("node-x").Label("zone", "zone2").Label("node", "node-x").Obj(),
+				st.MakeNode().Name("node-y").Label("zone", "zone2").Label("node", "node-y").Obj(),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("p-a1").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-a2").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b1").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-y1").Node("node-y").Label("foo", "").Obj(),
+				st.MakePod().Name("p-y2").Node("node-y").Label("foo", "").Obj(),
+			},
+			want: &preFilterState{
+				Constraints: []topologySpreadConstraint{
+					{
+						MaxSkew:            1,
+						TopologyKey:        "zone",
+						Selector:           mustConvertLabelSelectorAsSelector(t, st.MakeLabelSelector().In("foo", []string{"b"}).Label("foo", "a").Obj()),
```

Comment:
what does that mean? what exactly is the constraint? `foo in {b, a}`?

Reply:
It means like user set the same key in Selector and MatchLabelKeys at the same time. But the value in Selector isn't as the value get for pod.labels. Like this 
```yaml
kind: Pod
apiVersion: v1
metadata:
  name: mypod
  labels:
    foo: a  ---> foo=a
spec:
  topologySpreadConstraints:
  - maxSkew: 1
    topologyKey: zone
    whenUnsatisfiable: DoNotSchedule
    labelSelector:
      matchExpressions:   --> foo in {b}
      - key: foo
        operator: In
        values:
        - b
    matchLabelKeys:
    - foo   --> foo
  containers:
  - name: pause
    image: k8s.gcr.io/pause:3.1
```
If like this no pod will match this conditions.

Reply:
I meant how will that translate as a selector? two contradicting requirements?

Reply:
Yes. Selector will be translated as two contradicting requirements for this case.

like this ---->   foo in {b}; foo=a

Reply:
It's awkward, but users may indeed write as such by mistake, so add a UT to simulate

Reply:
As mentioned on the validation comment, lets just validate that keys are not shared between matchLabelKeys and labelSelector